docs/sql/datetime.md

```diff
@@ -262,12 +262,31 @@ computation on whole days.
 
 `DATE`, `TIME` and `TIMESTAMP` have no time zone.
 
-## Important unsupported operations
+## Non-deterministic functions
 
-Since DBSP is a *deterministic* query engine, it does not currently
-offer support for any function that depends on the current time.  So
-the following are *not* supported: `LOCALTIME`, `LOCALTIMESTAMP`,
-`CURRENT_TIME`, `CURRENT_DATE`, `CURRENT_TIMESTAMP`.
+Since DBSP is a *deterministic* query engine, it supports real-time
+based functions in way which is different from other SQL engines.
+Currently the only such function supported is `NOW`.  This function
+returns a TIMESTAMP value.  Programs that use the `NOW` function need
+to also declare the following table as the first table in the program:
+
+```sql
+CREATE TABLE NOW(now TIMESTAMP NOT NULL LATENESS INTERVAL 0 SECONDS);
+```
+
+(In the future this table will be automatically synthesized by the
+compiler, but for now its declaration must exist in the program.)
+
+The environment has to maintain the invariant that this table always
+contains a single value.  Moreover, deleting a value and inserting a
+new one requires the newly inserted value to be larger than the
```

Comment:
How would one do this? We don't have a way to delete/update values without a primary key.

Reply:
Http

Reply:
We can also build an adapter to do this automatically

Reply:
Have you tried this end to end? Because I don't think you can do this even through http.

Reply:
We should be able to delete from tables without a primary key just fine. Not sure what you're referring to, @lalithsuresh

Reply:
in json you can specify "insert" or "delete"

Reply:
How would you do this with HTTP? Read the value of NOW, then issue an update using the insert_delete format with {"delete": {"now": <last_read_value_of_now>}} ?

Reply:
You remove the value you just inserted previously.

Reply:
So the client either remembers the last value of now, or it has to read it from the view every time, which means the later has to be materialized too.

Reply:
Or we can build an adapter which does this automatically every time you call step.
But any of these will work.

Reply:
We can also add a primary key to this table.

Reply:
*that* would be very contrived